README.md

```diff
@@ -1,15 +1,89 @@
 
-**NOTE:** Jubilant is in very early stages of development. This is pre-alpha code. Our intention is to release a 1.0.0 version early to mid 2025.
+Jubilant is currently in pre-release or "beta" phase ([PyPI releases](https://pypi.org/project/jubilant/#history)). Our intention is to release a 1.0.0 final version in May 2025.
+
+[**Read the full documentation.**](https://canonical-jubilant.readthedocs-hosted.com/)
+
+
+## Using Jubilant
+
+Jubilant is published to PyPI, so you can install and use it with your favourite Python package manager:
+
+```
+$ pip install jubilant
+# or
+$ uv add jubilant
+```
+
+To use Jubilant in Python code, instantiate a `Juju` instance and call methods like `deploy` and `wait`:
+
+```python
+import jubilant
+
+juju = jubilant.Juju()
+juju.deploy('snappass-test')
+juju.wait(jubilant.all_active)
+```
+
+Below is an example integration test [from Jubilant itself](https://github.com/canonical/jubilant/blob/main/tests/integration/test_basic.py). This test uses the [`juju` pytest fixture](https://github.com/canonical/jubilant/blob/main/tests/integration/conftest.py), which adds a model during setup, runs your test with a `Juju` instance pointing at that model, and destroys the model during teardown:
+
+```python
+def test_deploy(juju: jubilant.Juju):
+    juju.deploy('snappass-test')             # Deploy the charm
+    status = juju.wait(jubilant.all_active)  # Wait till the app and unit are 'active'
+
+    # Hit the Snappass HTTP endpoint to ensure it's up and running.
+    address = status.apps['snappass-test'].units['snappass-test/0'].address
+    response = requests.get(f'http://{address}:5000/', timeout=10)
+    response.raise_for_status()
+    assert 'snappass' in response.text.lower()
+```
+
+You don't have to use [Pytest](https://docs.pytest.org/en/stable/) with Jubilant, but it's what we usually recommend. Pytest's fixtures and its simple `assert`-based approach make writing tests easy.
 
 
 ## Design goals
 
-- Match the Juju CLI. We try to ensure methods, argument names, and response field names match the Juju CLI and its responses, with minor exceptions (such as "application" being shortened to "app" in `Status` fields).
-- Simple API. Higher-level operations will be in helpers and fixtures, not the main `Juju` class.
-- No `async`. This was a "feature" of python-libjuju that isn't desired for integration tests.
+We designed Jubilant so it would:
+
+- Match the Juju CLI. Method, parameter, and response field names match the Juju CLI, with minor exceptions (such as "application" being shortened to "app").
+- Have a simple API. Higher-level operations will be in helpers and fixtures, not the main `Juju` class (the only exception being `Juju.wait`).
+- Not use `async`. This was a "feature" of python-libjuju that adds complexity and isn't needed for integration tests.
```

Comment:
Is it worth mentioning here that most Juju CLI commands return quickly and complete asynchronously in the background anyway? I feel that's the main reason why using Python async was the wrong choice - for example if you asynchronously call `deploy` when that completes the deploy isn't complete, just whatever initial validation and scheduling is done; you need to do some sort of waiting for the deploy to really complete.

Reply:
Good call, done.